code/game/objects/effects/temporary_visuals/effect_trail.dm

```diff
@@ -0,0 +1,45 @@
+/// An invisible effect which chases a target, spawning spikes every so often.
+/obj/effect/temp_visual/effect_trail
+	name = "effect trail"
+	desc = "An invisible effect, how did you examine this?"
+	icon = 'icons/mob/silicon/cameramob.dmi'
+	icon_state = "marker"
+	duration = 15 SECONDS
+	invisibility = INVISIBILITY_ABSTRACT
+	/// Typepath of our spawned effect
+	var/spawned_effect
+	/// How often do we spawn our other effect?
+	var/spawn_interval = 0.5 SECONDS
+	/// Speed at which we chase target
+	var/move_speed = 3
+	/// What are we chasing?
+	var/datum/weakref/target
+	/// Stop spawning if we have this many effects already
+	var/max_spawned = 20
+	/// Do we home in after we started moving?
+	var/homing = TRUE
+	/// Handles chasing the target
+	var/datum/move_loop/movement
+
+/obj/effect/temp_visual/effect_trail/Initialize(mapload, atom/target)
```

Comment:
ehhh feels kinda weird to have both the `weakref` of the target and the target be the same name... i wouldn't necessarily mind it if we stored a `var/atom/target` on the `effect_trail` type but since it's a weakref I think it should be named differently. i can get that we do the whole `src.target = WEAKREF(target)` but it just feels awk to have those two things be named the same when they are vastly different (where one is a datum subtype, and the other is a potential supertype for physical objects)

Reply:
Easier to just not make it a weakref actually, it's already registered to the deletion signal so I couldn't tell you why I made it one in the first place.